R/safe2x2Test.R

```diff
@@ -1,1148 +1,2016 @@
+    #log odds: theta A in (0,1), no reparameterization needed
+    thetaAvec <- rhoGrid
+  } else {
+    #if delta < 0, reparameterize + translate
+    thetaAvec <- rhoGrid*(1 - abs(delta)) - ifelse(delta < 0, delta, 0)
+  }
+  CurrentWorstCaseQuantile <- 0
+  CurrentWorstCasePower <- 1
 
-        calculateMarginalAlternative <-  function(h) {
-          exp(na1 * log(h[1]) + (na - na1) * log(1 - h[1]) + nb1 * log(h[2]) +
-                (nb -nb1) * log(1 - h[2]))
-        }
+  message(paste("Simulating E values and stopping times for divergence between groups of ", delta))
+  pbSafe <- utils::txtProgressBar(style=1)
+  for(t in seq_along(thetaAvec)){
+    StoppingTimes <- stopEs <- numeric(M)
 
-        #retrieve pbar1 at each time point (is a vector now!)
-        pbar1 <- rowSums(0.5 * apply(H1, 1, calculateMarginalAlternative))
+    thetaA <- thetaAvec[t]
+    if(restriction == "logOddsRatio"){
+      thetaB <- calculateThetaBFromThetaAandLOR(thetaA, delta)
+    } else {
+      thetaB <- thetaA + delta
+    }
 
-        #n is a vector now! And so is pbar0!
-        n <- na + nb
-        pbar0 <- 0.5 ^ n
+    for(i in 1:M){
+      #For every m, draw a sample of max streamlength and record the time
+      #at which we would have stopped
+      ya <- rbinom(n = maxSimStoptime, size = na, prob = thetaA)
+      yb <- rbinom(n = maxSimStoptime, size = nb, prob = thetaB)
+      simResult <- calculateSequential2x2E(aSample = ya, bSample = yb,
+                                                  priorValues = priorValues,
+                                                  restriction = restriction,
+                                                  #if explicitly passsed deltaDesign (neq delta), use that one for test
+                                                  #e.g. when studying effect of overestimated/ underestimated effect size
+                                                  delta = ifelse(is.null(deltaDesign), delta, deltaDesign),
+                                                  na = na,
+                                                  nb = nb,
+                                                  simSetting = TRUE,
+                                                  alphaSim = alpha)
+      StoppingTimes[i] <- simResult [["stopTime"]]
+      stopEs[i] <- simResult[["stopE"]]
+      utils::setTxtProgressBar(pbSafe, value=((t-1)*M+i)/(length(thetaAvec)*M))
+    }
 
-        if (sum(pbar1 / pbar0 >= 20) > 0) {
-          #at some time point, we rejected: record the first time point
-          rejected[i] <- TRUE
-          n.final[i] <- n[min(which(pbar1 / pbar0 >= 20))]
-        } else {
-          #we have not rejected and stopped collecting according to our stopping rule
-          rejected[i] <- FALSE
-          n.final[i] <- na.max + nb.max
-        }
-      }
+    #get the quantile for (1-b) power
+    CurrentQuantile <- ifelse(expectedStopTime, mean(StoppingTimes), quantile(StoppingTimes, probs = 1 - beta))
```

Comment:
Wellicht is dit bedoelt, maar het is goed om te weten dat ifelse soms apart gedrag vertoond en dat de output van dezelfde dimensie is als expectedStopTime

Reply:
waren allemaal 1d. Heb het aangepast maar nu 5 regels ipv 1.